### DISCUSSION TITLE
flashing killed my sonoff mini extreme (MINIR4)

### DISCUSSION THREAD

**Discussion by LachNed** [RavenSystem/esp-homekit-devices#2223] [Q&A] [answered]
I bought these very small new sonoff mini modules and flashed them with the NodeMCU PYflasher with fullhaaboot.bin and also tried fullhaaboot_esp32.bin with the same result... 
I used this programming rig (https://www.thingiverse.com/thing:6457602)
...after flashing finished successfully (see protocol below), the modules are dead... No LED, no WIFI, no nothing

Does anybody have the same problem or any idea how to solve this?
Did I killed all my modules?
Any help is appreciated

<img width="598" alt="image" src="https://github.com/RavenSystem/esp-homekit-devices/assets/81774728/a3b6b399-06f2-4178-ad55-1274da48b648">

The flashing protocol:

esptool.py v3.0
Serial port /dev/cu.usbserial-A50285BI
Connecting......
Detecting chip type... ESP32
Chip is ESP32-D0WD-V3 (revision 3)
Features: WiFi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
Crystal is 40MHz
MAC: e4:65:b8:19:7b:2c
Uploading stub...
Running stub...
Stub running...
Changing baud rate to 460800
Changed.
Configuring flash size...
Auto-detected Flash size: 4MB
Erasing flash (this may take a while)...
Chip erase completed successfully in 2.3s
Compressed 870784 bytes to 492663...
Wrote 870784 bytes (492663 compressed) at 0x00000000 in 11.5 seconds (effective 607.1 kbit/s)...
Hash of data verified.

Leaving...
Staying in bootloader.

Firmware successfully flashed. Unplug/replug or reset device 
to switch back to normal boot mode.

**Comment by hartzell** [comment]
You could try flashing Tasmota onto it and see if everything's working.  Once you make sense of it using *that* custom, non-Sonoff firmware you might have insights into getting it to work with HAA.

E.g. https://notenoughtech.com/home-automation/sonoff-minir4-extreme-released-and-flashed-with-tasmota/

**Reply by LachNed** [reply]
Thank you for the hints...
I already tried flashing Tasmota onto the module, but the result is the same...
No AP visible... I have no idea how to solve it and how I can get my modules running with HAA
Or should I use an HAA esp32-version for the sonoff mini4r ?

Here's the Log of flashing "tasmota.bin" without any errors:
_esptool.py v2.6
Serial port /dev/cu.usbserial-A50285BI
Connecting....
Detecting chip type... ESP32
Chip is ESP32D0WDQ5 (revision 1)
Features: WiFi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
MAC: e4:65:b8:1a:ac:80
Uploading stub...
Running stub...
Stub running...
Changing baud rate to 460800
Changed.
Configuring flash size...
Auto-detected Flash size: 4MB
Erasing flash (this may take a while)...
Chip erase completed successfully in 1.6s
Compressed 661632 bytes to 472673...
Wrote 661632 bytes (472673 compressed) at 0x00000000 in 10.9 seconds (effective 484.8 kbit/s)...
Hash of data verified.

Leaving...
Staying in bootloader.

Firmware successfully flashed. Unplug/replug or reset device 
to switch back to normal boot mode._

**Reply by hartzell** [reply]
It says:

> Chip is ESP32D0WDQ5 (revision 1)

so you should be using an ESP32 build, not `tasmota.bin`.

I'm a little fuzzy on the details, you should read up more on the [ESP32 page](https://tasmota.github.io/docs/ESP32/#esp32), but I suspect that if you're flashing via pogo-pins/hardware, then [tasmota32.factory.bin](https://ota.tasmota.com/tasmota32/release/tasmota32.factory.bin) is the one that should work for you.

**Reply by LachNed** [reply]
I also tried yesterday to flash [tasmota32.factory.bin](https://ota.tasmota.com/tasmota32/release/tasmota32.factory.bin) but with the same result... No AP visible...

**Reply by LachNed** [reply]
Good news
Last week I tried flashing again... now with 100% success. All my modules setup the access point with the latest HAA firmware. 
Don't know what's going wrong the least time. 
Thank you José and and thanks to all for this great project

This thread can be closed...

**Comment by RavenSystem** [comment, marked as answer]
You must flash fullhaaboot_esp32.bin file with the right commands: https://github.com/RavenSystem/esp-homekit-devices/wiki/installation#installing-haa

**Reply by LachNed** [reply]
First I want to thank you for this project and all your great work...

I flashed the sonoff with the esp32-version and ... surprise... an AP with HAA-1AAC00 appeared in the WIFI list...
I connected my iPhone to that AP and tried to access by 192.168.4.1:4567 ... BUT a few seconds later it disappeared and never comes back... 
An additional flash with the same settings doesn't help... the AP will not come back... not with FTDI power supply or main power supply. 
really frustrating....

**Reply by LachNed** [reply]
Are there any other ideas?
The flash progress always  finish with correct log information. I assume... the flash is not the problem.... The module will not boot with the flashed firmware...
I would like to send one of my Sonoff modules and the flashing jig (I used) to somebody who can analyse it to find the reasons for all my problems...
Is there anybody interested in such a job?
